boards/seeed/xiao_nrf54l15/doc/index.rst

```diff
@@ -0,0 +1,104 @@
+.. zephyr:board:: xiao_nrf54l15
+
+Overview
+********
+
+Seeed Studio XIAO nRF54L15 is a compact, high-performance development board featuring the cutting-edge
+Nordic nRF54L15 chip. This next-generation SoC integrates an ultra-low power multiprotocol 2.4 GHz
+radio and an MCU with a 128 MHz Arm® Cortex®-M33 processor and an Arm® Cortex®-M0+ for advanced
+power management. It offers scalable memory up to 1.5 MB NVM and 256 KB RAM.
+Designed for exceptional ultra-low power consumption, it significantly extends battery life.
+Its robust radio supports Bluetooth® 6.0 (including Channel Sounding), Matter, Thread, Zigbee,
+and a high-throughput 2.4 GHz proprietary mode (up to 4 Mbps). The board includes a comprehensive
+set of peripherals, an integrated 128 MHz RISC-V coprocessor, and advanced security features like
+TrustZone® isolation and cryptographic engine protection. With built-in lithium battery management,
+XIAO nRF54L15 is ideal for compact, secure, and energy-efficient IoT solutions such as smart wearables,
+industrial sensors, and advanced human-machine interfaces.
```

Comment:
add some breathing space instead of making it just one big, hard to read, paragraph
see https://builds.zephyrproject.io/zephyr/pr/92090/docs/boards/seeed/xiao_nrf54l15/doc/index.html